core/services/relay/evm/chain_reader.go

```diff
@@ -110,15 +110,18 @@ func (cr *chainReader) init(chainContractReaders map[string]types.ChainContractR
 	return nil
 }
 
-func (cr *chainReader) Start(_ context.Context) error {
+func (cr *chainReader) Start(ctx context.Context) error {
 	return cr.StartOnce("ChainReader", func() error {
-		return cr.contractBindings.ForEach(readBinding.Register)
+		return cr.contractBindings.ForEach(ctx, readBinding.Register)
 	})
 }
 
 func (cr *chainReader) Close() error {
 	return cr.StopOnce("ChainReader", func() error {
-		return cr.contractBindings.ForEach(readBinding.Unregister)
+		// TODO: Propagate context
```

Comment:
Can we link this to a ticket? Something like 

// TODO: https://smartcontract-it.atlassian.net/browse/BCF-999

Reply:
Actually this is a tricky case. Since we are already closing, we want to do this ASAP. The new background context seems necessary, but we should add a short timeout too.

Reply:
Bump. Let's drop the TODO and swap in a timeout - maybe ~1 second? IIRC our default overall shutdown limit is ~10s

Reply:
Sounds good